django/contrib/gis/geos/collections.py

```diff
@@ -53,7 +53,7 @@ def _create_collection(self, length, items):
             # allow GEOSGeometry types (python wrappers) or pointer types
             capi.geom_clone(getattr(g, 'ptr', g)) for g in items
         ])
-        return capi.create_collection(c_int(self._typeid), byref(geoms), c_uint(length))
+        return capi.create_collection(c_int(self._typeid), geoms, c_uint(length))
```

Comment:
Is calling `c_int()` and `c_uint()` redundant as well?

Reply:
Generally, I wanted to make minimal changes to the code as it hasn't been touched since 2009 (https://github.com/django/django/commit/66e1670efae34d721e374788e4c3f8b5fe5fa481), but I think your suggestions make sense!

Upon further investigation, calling a function calls from_param on each argtype, and `c_uint.from_param(c_uint(length))` (my current version, without removing the pre-cast to c_uint here) returns a c_uint instance, whereas `c_uint.from_param(length)` if we removed the pre-cast would return `<cparam 'I' 5>` which appears might be slightly more optimized.

I think the latter, making the change you suggested, would be both simpler and more performant - we should trust Python to optimize the FFI code and not over-pre-cast things just because that's how it was done previously. I'll remove the pre-casts alongside a change for the release note.

For reference, removing byref *is* necessary because while `POINTER(POINTER(...)).from_param(array_of_geom_pointers)` automatically casts, `POINTER(POINTER(...)).from_param(byref(array_of_geom_pointers))` throws a mismatched type error. Letting from_param do the heavy lifting seems to be a good trend here!

(Though on further investigation, it seems the byref issue above has been marked with an XXX in the Python source code since May 2010 and perhaps earlier... we are dredging up some ancient pre-Git horrors here! https://github.com/python/cpython/blob/3.10/Modules/_ctypes/_ctypes.c#L3993 Happy to continue to iterate in whatever way the core team thinks is best on all these fronts.)